src/zarr/core/metadata/v2.py

```diff
@@ -284,3 +284,19 @@ def parse_fill_value(fill_value: object, dtype: np.dtype[Any]) -> Any:
             raise ValueError(msg) from e
 
     return fill_value
+
+
+def default_fill_value(dtype: np.dtype[Any]) -> Any:
```

Comment:
The equivalent method in v3 was made private in #2036.
```suggestion
def _default_fill_value(dtype: np.dtype[Any]) -> Any:
```